### PR TITLE
[DPE-4369] SQL queries exposing passwords on postgresql logs

### PR DESCRIPTION
**Pull request by lucasgameiroborges**
## Issue

config value `log_statement = 'ddl'` logs queries such as `ALTER USER/ROLE`, `CREATE USER/ROLE` that expose users passwords on log files.

## Solution

set `log_statement='none'` when submitting sensitive queries:

```
BEGIN;
SET LOCAL log_statement = 'none';
CREATE ROLE "backup" WITH LOGIN SUPERUSER ENCRYPTED PASSWORD 'MRiT223dummy2322nHESx';
...
COMMIT;
```

## Test

New integration test failing before the change: https://github.com/canonical/postgresql-k8s-operator/actions/runs/9483057657/job/26129752923#step:21:386
... and passing after the change: https://github.com/canonical/postgresql-k8s-operator/actions/runs/9483490912/job/26131217890?pr=506#step:21:182

### CODE REVIEW

**Review by marceloneppel — APPROVED** [review]
LGTM! For the failure in `test_new_relation.py` I think you can replicate the https://github.com/canonical/postgresql-k8s-operator/pull/491/commits/9233cb786caeb43fba0e985e3898e83eba1b2459 commit until https://github.com/canonical/redis-k8s-operator/issues/87 is fixed.

File: lib/charms/postgresql_k8s/v0/postgresql.py
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:1]: Should bump LIBPATCH.

**Review by dragomirp — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Tnx!

**Comment by reneradoi** [comment]
@lucasgameiroborges The [bug](https://github.com/canonical/redis-k8s-operator/issues/87) in `redis-k8s-operator` has been fixed.

**Pull request merged**